python/trifinger_simulation/collision_objects.py

```diff
@@ -145,3 +138,205 @@ def __del__(self):
+                shapeType=pybullet.GEOM_CAPSULE,
+                radius=radius,
+                height=height,
+                **self._kwargs,
+            )
+        elif type == "cylinder":
+            self.object_id = pybullet.createCollisionShape(
+                shapeType=pybullet.GEOM_CYLINDER,
+                radius=radius,
+                height=height,
+                **self._kwargs,
+            )
+        elif type == "custom":
+            if mesh_file_path is None:
+                raise ValueError(
+                    "Please pass a mesh file to import the custom"
+                    "desired object from. :)"
+                )
+            if is_concave:
+                flags = pybullet.GEOM_FORCE_CONCAVE_TRIMESH
+            else:
+                flags = 0
+            self.object_id = pybullet.createCollisionShape(
+                shapeType=pybullet.GEOM_MESH,
+                fileName=mesh_file_path,
+                flags=flags,
+                **self._kwargs,
+            )
+        self.object = pybullet.createMultiBody(
+            baseCollisionShapeIndex=self.object_id,
+            baseVisualShapeIndex=-1,
+            basePosition=position,
+            baseOrientation=orientation,
+            baseMass=mass,
+            **self._kwargs,
+        )
+
+        # sets visual properties of the object
+        if color_rgba is not None:
+            self.change_texture(color_rgba=color_rgba)
+
+        # sets dynamics of the object
+        self.change_dynamics(
+            lateral_friction=lateral_friction,
+            spinning_friction=spinning_friction,
+            restitution=restitution,
+        )
+
+    def change_dynamics(
+        self, lateral_friction=1, spinning_friction=0.001, restitution=0
+    ):
+        """Allows to change the dynamics properties of the object.
+
+        Args:
+
+            mass (float): how heavy should this object be
+            lateral_friction (double): the friction coefficient of the object
+            spinning_friction (double): the spinning friction of the object
+                around its contact normal
+            restitution (double): the coefficient of restittuion for the object
+        """
+        pybullet.changeDynamics(
+            bodyUniqueId=self.object,
+            linkIndex=-1,
+            lateralFriction=lateral_friction,
+            spinningFriction=spinning_friction,
+            restitution=restitution,
+            **self._kwargs,
+        )
+
+    def change_texture(self, color_rgba=None, texture_file_path=None):
+        """Changes the texture and color of the object.
+
+        Allows to load some custom image or texture onto the object, for eg.
+        this could be a png file. It also allows to change the object colour.
+        Note that, in case both desired color and desired texture are passed,
+        then both will get loaded onto the object, and the texture will have
```

Comment:
> then both will get loaded onto the object

The current implementation will not set the colour if both are specified. (see also comment on equivalent method in SimFinger).